src/gvizdoom_tests/TestActions.cpp

```diff
@@ -0,0 +1,51 @@
+#include <gtest/gtest.h>
+#include <gvizdoom/DoomGame.hpp>
+
+#include "Action.hpp"
+#include "TestUtils.hpp"
+
+#include <opencv2/core/mat.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/imgproc.hpp>
+
+using namespace gvizdoom;
+
+TEST(TestActions, AllActions)
+{
+    auto& doomGame = DoomGame::instance();
+    GameConfig gameConfig{0, nullptr, false, 640, 480, true, GameConfig::HUD_STATUSBAR, 2, 3, 1, 4};
+    gameConfig.videoWidth = 640;
+    gameConfig.videoHeight = 480;
+    gameConfig.videoTrueColor = true;
+    doomGame.init(gameConfig);
+
+    std::vector<Action> actions;
+    
+    for (size_t i = 0; i < 50; ++i)
+        actions.emplace_back(Action::Key::ACTION_ATTACK, 0);
+
+    for (size_t i = 0; i < 50; ++i)
+        actions.emplace_back(static_cast<int>(Action::Key::ACTION_FORWARD | Action::Key::ACTION_ATTACK), 100);
+
+    for (size_t i = 0; i < 50; ++i)
+        actions.emplace_back(Action::Key::ACTION_FORWARD, 0);
+
+
+    for (const auto& a : actions)
+        doomGame.update(a);
+    
+    cv::Mat screenPixels(doomGame.getScreenHeight(),
+            doomGame.getScreenWidth(),
+            CV_8UC4,
+            const_cast<uint8_t*>(doomGame.getPixelsBGRA()));
+
+    // This will inevitably assume that the tests are run in {pwd}/foo folder
+    const cv::Mat screenPixelsExpected{cv::imread("../test_images/testactions_allactions_expected.png")};
+    ASSERT_EQ(screenPixelsExpected.empty(), false);
+    cv::cvtColor(screenPixels, screenPixels, cv::COLOR_BGRA2BGR);
+    const cv::Mat diff = screenPixelsExpected - screenPixels;
```

Comment:
Beware of negative values! They will reduce the error. I think what you want is `cv::absdiff`.